test_decimals_include_nan is flaky

```
---------------------------------- Hypothesis ----------------------------------
You can add @seed(327206777863050896557169937789875330632) to this test or run pytest with --hypothesis-seed=327206777863050896557169937789875330632 to reproduce this failure.
__________________________ test_decimals_include_nan ___________________________
[gw8] linux -- Python 3.6.3 /usr/bin/python
tests/cover/test_numerics.py:88: in test_decimals_include_nan
    @given(decimals())
tests/common/utils.py:64: in inverted_test
    f(*arguments, **kwargs)
tests/cover/test_numerics.py:88: in test_decimals_include_nan
    @given(decimals())
E   hypothesis.errors.MultipleFailures: Hypothesis found 2 distinct failures.
---------------------------------- Hypothesis ----------------------------------
Falsifying example: test_decimals_include_nan(x=Decimal('sNaN'))
Traceback (most recent call last):
  File "/build/python-hypothesis/src/hypothesis-python-3.44.1/src/hypothesis/core.py", line 818, in run
    falsifying_example.__expected_traceback,
  File "/build/python-hypothesis/src/hypothesis-python-3.44.1/src/hypothesis/core.py", line 579, in execute
    result = self.test_runner(data, run)
  File "/build/python-hypothesis/src/hypothesis-python-3.44.1/src/hypothesis/executors.py", line 58, in default_new_style_executor
    return function(data)
  File "/build/python-hypothesis/src/hypothesis-python-3.44.1/src/hypothesis/core.py", line 571, in run
    return test(*args, **kwargs)
  File "/build/python-hypothesis/src/hypothesis-python-3.44.1/tests/cover/test_numerics.py", line 88, in test_decimals_include_nan
    @given(decimals())
  File "/build/python-hypothesis/src/hypothesis-python-3.44.1/src/hypothesis/core.py", line 518, in test
    result = self.test(*args, **kwargs)
  File "/build/python-hypothesis/src/hypothesis-python-3.44.1/tests/cover/test_numerics.py", line 90, in test_decimals_include_nan
    assert not math.isnan(x)
  File "/usr/lib/python3.6/_pydecimal.py", line 1651, in __float__
    raise ValueError("Cannot convert signaling NaN to float")
ValueError: Cannot convert signaling NaN to float

Falsifying example: test_decimals_include_nan(x=Decimal('NaN'))
Traceback (most recent call last):
  File "/build/python-hypothesis/src/hypothesis-python-3.44.1/src/hypothesis/core.py", line 818, in run
    falsifying_example.__expected_traceback,
  File "/build/python-hypothesis/src/hypothesis-python-3.44.1/src/hypothesis/core.py", line 579, in execute
    result = self.test_runner(data, run)
  File "/build/python-hypothesis/src/hypothesis-python-3.44.1/src/hypothesis/executors.py", line 58, in default_new_style_executor
    return function(data)
  File "/build/python-hypothesis/src/hypothesis-python-3.44.1/src/hypothesis/core.py", line 571, in run
    return test(*args, **kwargs)
  File "/build/python-hypothesis/src/hypothesis-python-3.44.1/tests/cover/test_numerics.py", line 88, in test_decimals_include_nan
    @given(decimals())
  File "/build/python-hypothesis/src/hypothesis-python-3.44.1/src/hypothesis/core.py", line 518, in test
    result = self.test(*args, **kwargs)
  File "/build/python-hypothesis/src/hypothesis-python-3.44.1/tests/cover/test_numerics.py", line 90, in test_decimals_include_nan
    assert not math.isnan(x)
AssertionError: assert not True
 +  where True = <built-in function isnan>(Decimal('NaN'))
 +    where <built-in function isnan> = math.isnan
```

--- a/tests/cover/test_numerics.py
+++ b/tests/cover/test_numerics.py
@@ -88,10 +88,7 @@ def test_decimals_include_nan():
 
 
 def test_decimals_include_inf():
-    find_any(
-        decimals(),
-        lambda x: assume(not x.is_snan()) and x.is_infinite()
-    )
+    find_any(decimals(), lambda x: x.is_infinite())
 
 
 @given(decimals(allow_nan=False))

--- a/tests/cover/test_numerics.py
+++ b/tests/cover/test_numerics.py
@@ -24,7 +24,7 @@ import pytest
 
 from hypothesis import given, assume, reject
 from hypothesis.errors import InvalidArgument
-from tests.common.utils import fails
+from tests.common.debug import find_any
 from hypothesis.strategies import data, none, tuples, decimals, integers, \
     fractions
 
@@ -72,11 +72,11 @@ def test_fuzz_decimals_bounds(data):
         assert val.as_tuple().exponent == -places
 
 
-@fails
-@given(decimals())
-def test_all_decimals_can_be_exact_floats(x):
-    assume(x.is_finite())
-    assert decimal.Decimal(float(x)) == x
+def test_all_decimals_can_be_exact_floats():
+    find_any(
+        decimals(),
+        lambda x: assume(x.is_finite()) and decimal.Decimal(float(x)) == x
+    )
 
 
 @given(fractions(), fractions(), fractions())
@@ -84,17 +84,15 @@ def test_fraction_addition_is_well_behaved(x, y, z):
     assert x + y + z == y + x + z
 
 
-@fails
-@given(decimals())
-def test_decimals_include_nan(x):
-    assert not math.isnan(x)
+def test_decimals_include_nan():
+    find_any(decimals(), lambda x: x.is_nan())
 
 
-@fails
-@given(decimals())
-def test_decimals_include_inf(x):
-    assume(not x.is_snan())
-    assert not math.isinf(x)
+def test_decimals_include_inf():
+    find_any(
+        decimals(),
+        lambda x: assume(not x.is_snan()) and math.isinf(x)
+    )
 
 
 @given(decimals(allow_nan=False))

--- a/tests/cover/test_numerics.py
+++ b/tests/cover/test_numerics.py
@@ -17,7 +17,6 @@
 
 from __future__ import division, print_function, absolute_import
 
-import math
 import decimal
 
 import pytest
@@ -91,19 +90,18 @@ def test_decimals_include_nan():
 def test_decimals_include_inf():
     find_any(
         decimals(),
-        lambda x: assume(not x.is_snan()) and math.isinf(x)
+        lambda x: assume(not x.is_snan()) and x.is_infinite()
     )
 
 
 @given(decimals(allow_nan=False))
 def test_decimals_can_disallow_nan(x):
-    assert not math.isnan(x)
+    assert not x.is_nan()
 
 
 @given(decimals(allow_infinity=False))
 def test_decimals_can_disallow_inf(x):
-    assume(not x.is_snan())
-    assert not math.isinf(x)
+    assert not x.is_infinite()
 
 
 @pytest.mark.parametrize('places', range(10))

